spacy/tokens/span.pyx

```diff
@@ -469,30 +469,6 @@ cdef class Span:
         """
         return "".join([t.text_with_ws for t in self])
 
-    @property
-    def noun_chunks(self):
-        """Yields base noun-phrase `Span` objects, if the document has been
-        syntactically parsed. A base noun phrase, or "NP chunk", is a noun
-        phrase that does not permit other NPs to be nested within it – so no
-        NP-level coordination, no prepositional phrases, and no relative
-        clauses.
-
-        YIELDS (Span): Base noun-phrase `Span` objects.
-
-        DOCS: https://nightly.spacy.io/api/span#noun_chunks
-        """
-        # Accumulate the result before beginning to iterate over it. This
-        # prevents the tokenisation from being changed out from under us
-        # during the iteration. The tricky thing here is that Span accepts
-        # its tokenisation changing, so it's okay once we have the Span
-        # objects. See Issue #375
-        spans = []
-        cdef attr_t label
-        if self.doc.noun_chunks_iterator is not None:
-            for start, end, label in self.doc.noun_chunks_iterator(self):
```

Comment:
Wait I think we do want this, although the implementation looks broken to me?

I think the idea is that it should get just the noun chunks within the span. But I can't see how this code would work, isn't the `start` and `end` here wrong?

The functionality is useful to harmonize the `Doc` and `Span` APIs, and to get all noun chunks within a sentence.